nautobot/core/views/__init__.py

```diff
@@ -14,93 +15,38 @@
 from packaging import version
 from graphene_django.views import GraphQLView
 
-from nautobot.circuits.models import Circuit, Provider
-from nautobot.dcim.models import (
-    Cable,
-    ConsolePort,
-    Device,
-    DeviceType,
-    Interface,
-    PowerPanel,
-    PowerFeed,
-    PowerPort,
-    Rack,
-    Site,
-    VirtualChassis,
-)
 from nautobot.core.constants import SEARCH_MAX_RESULTS, SEARCH_TYPES
 from nautobot.core.forms import SearchForm
 from nautobot.core.releases import get_latest_release
-from nautobot.extras.choices import JobResultStatusChoices
-from nautobot.extras.models import GitRepository, GraphQLQuery, ObjectChange, JobResult
+from nautobot.extras.models import GraphQLQuery
+from nautobot.extras.registry import registry
 from nautobot.extras.forms import GraphQLQueryForm
-from nautobot.ipam.models import Aggregate, IPAddress, Prefix, VLAN, VRF
-from nautobot.tenancy.models import Tenant
-from nautobot.virtualization.models import Cluster, VirtualMachine
 
 
 class HomeView(TemplateView):
     template_name = "home.html"
 
-    def get(self, request):
+    def render_additional_content(self, request, context, details):
+        # Collect all custom data using callback functions.
+        for key, data in details.get("custom_data", {}).items():
+            if callable(data):
+                context[key] = data(request)
+            else:
+                context[key] = data
 
-        connected_consoleports = (
-            ConsolePort.objects.restrict(request.user, "view")
-            .prefetch_related("_path")
-            .filter(_path__destination_id__isnull=False)
-        )
-        connected_powerports = (
-            PowerPort.objects.restrict(request.user, "view")
-            .prefetch_related("_path")
-            .filter(_path__destination_id__isnull=False)
-        )
-        connected_interfaces = (
-            Interface.objects.restrict(request.user, "view")
-            .prefetch_related("_path")
-            .filter(_path__destination_id__isnull=False, pk__lt=F("_path__destination_id"))
-        )
+        # Create standalone template
+        path = f'{details["template_path"]}{details["custom_template"]}'
+        if os.path.isfile(path):
+            with open(path, "r") as f:
+                html = f.read()
+        else:
+            html = details["custom_template"].strip()
```

Comment:
So `custom_template` can be either a file path or a raw HTML template string? Probably would be cleaner to have two different keys here, perhaps `custom_template` and `custom_template_file`?

Reply:
Do you think its worth having both features? Or should i just keep the file?

Reply:
Good point. If all we're using presently is the file, let's just keep that.

Reply:
Going to remove the `else:` case here then?

Reply:
Removed